generator/src/main/kotlin/com/android/gradle/replicator/generator/project/GradleProjectGenerator.kt

```diff
@@ -213,13 +220,20 @@ class GradleProjectGenerator(
     ) {
         val metadataFile = folder.join("resource-metadata.json")
 
-        val metadataJson = JsonObject()
-        module.androidResources?.resourceMap?.also {
-            metadataJson.add("androidResources", Gson().toJsonTree(it))
-        }
-        metadataJson.addProperty("javaResources", module.javaResources?.fileCount ?: 0)
+        with(JsonWriter(FileWriter(metadataFile))) {
+            this.setIndent("  ")
+            beginObject()
+            module.androidResources?.also {
+                name("androidResources")
+                AndroidResourcesAdapter().write(this, it)
+            }
 
-        metadataFile.writeBytes(GsonBuilder().setPrettyPrinting().create().toJson(metadataJson).toByteArray())
+            name("javaResources")
+            SourceFilesAdapter().write(this, module.javaResources ?: DefaultSourceFilesInfo(0))
+
+            endObject()
```

Comment:
this should be in the same method as the beginObject. Move one of them to be co-located.

Reply:
This is in the same method as beginObject

`
    private fun generateModuleResourceMetadata(
            folder: File,
            module: ModuleInfo
    ) {
        val metadataFile = folder.join("resource-metadata.json")

        with(JsonWriter(FileWriter(metadataFile))) {
            this.setIndent("  ")
            beginObject()
            module.androidResources?.let {
                name("androidResources")
                AndroidResourcesAdapter().write(this, it)
            }

            name("javaResources")
            SourceFilesAdapter().write(this, module.javaResources ?: DefaultSourceFilesInfo(0))

            endObject()
            this.flush()
        }
    }`